docs/source/manual/servlets.rst

```diff
@@ -14,12 +14,52 @@ HealthCheckServlet
 ``HealthCheckServlet`` responds to ``GET`` requests by running all the [health checks](#health-checks)
 and returning ``501 Not Implemented`` if no health checks are registered, ``200 OK`` if all pass, or
 ``500 Internal Service Error`` if one or more fail. The results are returned as a human-readable
-``text/plain`` entity.
+``application/json`` entity.
+
+Configuration
+-------------
+
+``HealthCheckServlet`` supports the following configuration items.
+
+Servlet Context
+~~~~~~~~~~~~~~~
 
 ``HealthCheckServlet`` requires that the servlet context has a ``HealthCheckRegistry`` named
 ``com.codahale.metrics.servlets.HealthCheckServlet.registry``. You can subclass
-``MetricsServletContextListener``, which will add a specific ``HealthCheckRegistry`` to the servlet
-context.
+``HealthCheckServlet.ContextListener``, which will add a specific ``HealthCheckRegistry`` to the
+servlet context.
+
+An instance of ``ExecutorService`` can be provided via the servlet context using the name
+``com.codahale.metrics.servlets.HealthCheckServlet.executor``; by default, no thread pool is used to
+execute the health checks.
+
+An instance of ``HealthCheckFilter`` can be provided via the servlet context using the name
+``com.codahale.metrics.servlets.HealthCheckServlet.healthCheckFilter``; by default, no filtering is
+enabled. The filter is used to determine which health checks to include in the health status.
+
+An instance of ``ObjectMapper`` can be provided via the servlet context using the name
+``com.codahale.metrics.servlets.HealthCheckServlet.mapper``; if none is provided, a default instance
+will be used to convert the health check results to JSON.
+
+Initialization Parameters
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+``HealthCheckServlet`` supports the following initialization parameters:
+
+* ``com.codahale.metrics.servlets.HealthCheckServlet.httpStatusIndicator``: Provides the
+default setting that determines whether the HTTP status code is used to determine whether the
+application is healthy. If not provided, it defaults to ``"true"``
+
+Query Parameters
+~~~~~~~~~~~~~~~~
+
+``HealthCheckServlet`` supports the following query parameters:
+
+* ``httpStatusIndicator`` (`Boolean``): Determines whether the HTTP status code is used to
+  determine whether the application is healthy; if not provided, it defaults to the value from the
+  initialization parameter
```

Comment:
Could we describe the possible HTTP response codes (200, 500, 501)?

Reply:
They're in the lead paragraph; I split them out for clarity